{
  TestFile testDir=dist.getTestDir();
  testDir.file("src/main/java/AppException.java").writelns("public class AppException extends Exception { }");
  testDir.file("src/test/java/SomeTest.java").writelns("public class SomeTest {","@org.junit.Test public void pass() { }","@org.junit.Test public void fail() { org.junit.Assert.fail(); }","@org.junit.Test public void knownError() { throw new RuntimeException(); }","@org.junit.Test public void unknownError() throws AppException { throw new AppException(); }","}");
  testDir.file("build.gradle").writelns("usePlugin 'java'","dependencies { testCompile 'junit:junit:4.4' }","def listener = new TestListenerImpl()","test.addTestListener(listener)","test.ignoreFailures = true","class TestListenerImpl implements TestListener {","void suiteStarting(TestListener.Suite suite) { println 'START SUITE ' + suite.name }","void suiteFinished(TestListener.Suite suite) { println 'FINISH SUITE ' + suite.name }","void testStarting(TestListener.Test test) { println 'START TEST ' + test.name }","void testFinished(TestListener.Test test, TestListener.Result result) { println 'FINISH TEST ' + test.name + ' error: ' + result.error }","}");
  ExecutionResult result=executer.withTasks("test").run();
  assertThat(result.getOutput(),containsLine("START SUITE SomeTest"));
  assertThat(result.getOutput(),containsLine("FINISH SUITE SomeTest"));
  assertThat(result.getOutput(),containsLine("START TEST pass(SomeTest)"));
  assertThat(result.getOutput(),containsLine("FINISH TEST pass(SomeTest) error: null"));
  assertThat(result.getOutput(),containsLine("START TEST fail(SomeTest)"));
  assertThat(result.getOutput(),containsLine("FINISH TEST fail(SomeTest) error: junit.framework.AssertionFailedError: "));
  assertThat(result.getOutput(),containsLine("START TEST knownError(SomeTest)"));
  assertThat(result.getOutput(),containsLine("FINISH TEST knownError(SomeTest) error: java.lang.RuntimeException"));
  assertThat(result.getOutput(),containsLine("START TEST unknownError(SomeTest)"));
  assertThat(result.getOutput(),containsLine("FINISH TEST unknownError(SomeTest) error: java.lang.RuntimeException: AppException: null"));
}
